components/concept/storage/src/main/java/mozilla/components/concept/storage/BookmarksStorage.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.storage
+
+import mozilla.appservices.places.BookmarkFolder
+import mozilla.appservices.places.BookmarkItem
+import mozilla.appservices.places.BookmarkRoot
+import mozilla.appservices.places.BookmarkTreeNode
+import mozilla.appservices.places.BookmarkUpdateInfo
```

Comment:
This doesn't follow our current pattern of having `concept` components not depend on anything. This class is supposed to define simple interfaces, without imposing any external dependencies or implementations. What you have currently will force an application that only wants to consume an in-memory history storage component to actually depend on the libplaces native libraries!

This is the reason why `concept-sync` and `concept-storage` redefine some of the types exposed by libplaces, and provide conversion points (see the conventional `Types.kt`). You'll have to follow the same pattern here.

Reply:
Done